solutions/cf-616D.mdx

```diff
@@ -2,20 +2,67 @@
 id: cf-616D
 source: CF
 title: Longest k-Good Segment
-author: Sofia Yang
+author: Sofia Yang, Mohammad Nour Massri
 ---
 
 [Official Editorial](https://codeforces.com/blog/entry/22712)
+## Implementation
 
+**Time Complexity:** $\mathcal{O}(N)$
 <LanguageSection>
+<CPPSection>
+```cpp 
 
-<JavaSection>
+#include <bits/stdc++.h>
+using namespace std;
+int n,k;
+int a[500009];
+int frequency[1000009];//frequency of each number in the window
```

Comment:
I think it would be better to use a constant variable instead of hard-coding the array size. More aesthetic and less prone to potential errors.